app/src/main/java/com/example/triptracker/view/map/MapPopups.kt

```diff
@@ -202,28 +201,51 @@ fun PathItem(pinnedPlace: Pin, onClick: (Pin) -> Unit) {
   Spacer(modifier = Modifier.height(16.dp))
 }
 
-@SuppressLint("SuspiciousIndentation")
 @Composable
-fun AddressText(mpv: MapPopupViewModel, latitude: Float, longitude: Float) {
-  val address by mpv.address.observeAsState("Loading address...")
-  Log.d("AddressText", "Address: $address")
+fun AddressText(mapPopupViewModel: MapPopupViewModel, latitude: Float, longitude: Float) {
+  val address = remember { mutableStateOf("Loading address...") }
 
-  // take only the first part of the address (the street)
-  val addressParts = address.split(",")
-  val street = addressParts[0]
-
-  // Trigger the address fetch
-  LaunchedEffect(key1 = latitude, key2 = longitude) { mpv.fetchAddressForPin(latitude, longitude) }
+  // Launch a coroutine to fetch the address
+  LaunchedEffect(Unit) {
+    while (address.value == "Loading address...") {
+      mapPopupViewModel.fetchAddressForPin(latitude, longitude)
+      // Simulate a delay for fetching the address
+      delay(1000)
```

Comment:
Isn't there a better way to do this? I mean it works but not sure it's the proper way to do it. 
Or maybe just put a constant variable called DELAY or something like that.
If there isn't then I guess it's fine

Reply:
The thing is we know that the device is connected to the internet. And so we know it will fetch but not directly so it will try maybe 2 times. This delay is here to not send 1000000 requests, like 2-3 is sufficient in 99,9999% cases.

Reply:
This causes the app to crash unfortunately when you swipe very fast